### PR TITLE
Adapt to coq/coq#18973.

### PR DESCRIPTION
**Pull request by rlepigre**

### CODE REVIEW

File: serlib/ser_pstring.ml [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+
+open Sexplib.Std
+open Ppx_hash_lib.Std.Hash.Builtin
+open Ppx_compare_lib.Builtin
+
+module PierceSpec = struct
+  type t = Pstring.t
+  type _t = string [@@deriving sexp,yojson,hash,compare]
+end
+
```
Comment (ejgallego) [inline serlib/ser_pstring.ml:27]: Why do you pierce here, I see the Coq type is `Pstring.t = string` ?

Comment (ejgallego) [reply serlib/ser_pstring.ml:27]: Piercing is `Obj.magic` so it should not be used unless it is the last resort.

**Comment by ejgallego** [comment]
@rlepigre please trigger the coq-ci again for coq-lsp and after, for serapi (as it will need to use the new artifacts)

**Comment by proux01** [comment]
Upstream merged, please merge

**Pull request merged**